packages/pluginsdk/scripts/scaffold.js

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env node
+const path = require('path');
+const fs = require('fs');
+const child_process = require('child_process');
+const readlineSync = require('readline-sync');
+
+const pluginNameQuestion = 'Enter the short name for your plugin (default: myplugin): ';
+const pluginName = readlineSync.question(pluginNameQuestion, { defaultInput: 'myplugin' });
+
+const dirQuestion = `Directory to scaffold into (default: ${pluginName}-deck): `;
+const scaffoldTargetDir = readlineSync.question(dirQuestion, { defaultInput: `${pluginName}-deck` });
+
+// Rename these files
+const fileMapping = {
+  'scaffold.tsconfig.json': 'tsconfig.json',
+  'scaffold.prettierrc.js': '.prettierrc.js',
+  'scaffold-deck.gradle': `${scaffoldTargetDir}.gradle`,
+};
+
+const renamedFile = file => fileMapping[file] || file;
+
+function mkdirIfNotExists(dirPath) {
+  if (!fs.existsSync(dirPath)) {
+    fs.mkdirSync(dirPath);
+  }
+}
+
+function recursiveCopy(srcDir, targetDir) {
+  const contents = fs.readdirSync(srcDir);
+  contents
+    .filter(x => x !== '.' && x !== '..')
+    .forEach(file => {
+      const srcPath = path.resolve(srcDir, file);
+      const targetPath = path.resolve(targetDir, renamedFile(file));
+      const stat = fs.statSync(srcPath);
+      if (stat.isDirectory()) {
+        mkdirIfNotExists(targetPath);
+        recursiveCopy(srcPath, targetPath);
+      } else if (stat.isFile()) {
+        fs.writeFileSync(targetPath, fs.readFileSync(srcPath));
+      }
+    });
+}
+
+function updatePackageJson(pkgJsonPath, scaffoldPath) {
```

Comment:
here we update the `package.json` with the package name (as entered by the user) and also pin the dependency on `@spinnaker/pluginsdk` to the version doing the scaffolding.